File: pedigree-puzzle-generator/node/src/inheritance.ts
/**
 * Inheritance-mode enum and the associated biological-sex enum.
 *
 * Kept separate from puzzleBlueprint so the pure domain vocabulary
 * has nothing to do with pedigree-building mechanics -- answerExplainer
 * and modeHeuristics both depend on this, but not on each other.
 */

export const MODES = ['AD', 'AR', 'XLR', 'XLD', 'MT'] as const;
export type Mode = (typeof MODES)[number];

export type Sex = 'female' | 'male';

const FULL_NAMES: Readonly<Record<Mode, string>> = {
  AD: 'Autosomal Dominant',
  AR: 'Autosomal Recessive',
  XLR: 'X-linked Recessive',
  XLD: 'X-linked Dominant',
  MT: 'Mitochondrial',
};

export function modeFullName(mode: Mode): string {
  return FULL_NAMES[mode];
}

File: pedigree-puzzle-generator/python/src/pedigree_puzzle/inheritance.py
"""Inheritance-mode enum and the associated biological-sex enum.

Kept separate from :mod:`puzzle_blueprint` so the pure domain vocabulary
has nothing to do with pedigree-building mechanics -- answer_explainer
and mode_heuristics both depend on this, but not on each other.
"""

from __future__ import annotations

from enum import Enum


class Mode(str, Enum):
    AD = "AD"
    AR = "AR"
    XLR = "XLR"
    XLD = "XLD"
    MT = "MT"

    @property
    def full_name(self) -> str:
        return _FULL_NAMES[self]


class Sex(str, Enum):
    MALE = "male"
    FEMALE = "female"


_FULL_NAMES: dict[Mode, str] = {
    Mode.AD: "Autosomal Dominant",
    Mode.AR: "Autosomal Recessive",
    Mode.XLR: "X-linked Recessive",
    Mode.XLD: "X-linked Dominant",
    Mode.MT: "Mitochondrial",
}
